etcd/cluster-etcd-operator.md

```diff
@@ -0,0 +1,331 @@
+    MemberUnknown MemberConditionType = "Unknown"
+    // Degraded indicates the member pod is in a degraded state and should be restarted
+    MemberDegraded MemberConditionType = "Degraded"
+    // Remove indicates the member should be removed from the cluster
+    MemberRemove MemberConditionType = "Remove"
+    // MemberAdd is a member who is ready to join cluster but currently has not
+    MemberAdd MemberConditionType = "Add"
+)
+```
+
+Etcds: Actions to reconcile cluster membership are based of the observations reported by the configobservation controller.
+
+The example below shows an observed member failure caused by catastrophic failure. The result of the containers readiness probe failing the member is taken from the members bucket and placed into pending bucket `( Degraded = true, Progressing = true )`. After further observation it is confirmed that the container is actively in `CrashLoopBackOff`. The result of these observations is a new status of `Remove`. The remove status is then acted upon by the `clustermembercontroller` which removes the member from the etcd cluster by calling `MemberRemove` against the etcd `Cluster` API. The `staticpod` controller will also respond by stopping the static Pod, removing the existing data-dir and then starting the static Pod. The discovery init container then will wait for status now observed as `Unknown` to change to Add before it continues. `configobservation` controller observes the Pod and confirms etcd-member container is no longer CrashLooping and sets status to Add after which the member is added back to the etcd cluster with appropriate values populated for `ETCD_INITIAL_CLUSTER` allowing it join. Now readiness probe will succeed and member is added back to the member bucket `( Degraded = false, Progressing = false )`.
+```
+apiVersion: v1
+items:
+- apiVersion: operator.openshift.io/v1
+  kind: Etcd
+  metadata:
+    annotations:
+      release.openshift.io/create-only: "true"
+    creationTimestamp: "2019-10-02T12:38:41Z"
+    generation: 2
+    name: cluster
+    resourceVersion: "108019"
+    selfLink: /apis/operator.openshift.io/v1/etcds/cluster
+    uid: 905931e7-e511-11e9-8e06-020bc572eaf2
+  spec:
+    forceRedeploymentReason: ""
+    logLevel: ""
+    managementState: Managed
+    observedConfig:
+      cluster:
+        members:
+        - name: etcd-member-ip-10-0-154-109.us-east-2.compute.internal
+          peerURLs: https://etcd-0.sbatsche.devcluster.openshift.com:2380
+          status: Ready
+        - name: etcd-member-ip-10-0-164-65.us-east-2.compute.internal
+          peerURLs: https://etcd-2.sbatsche.devcluster.openshift.com:2380
+          status: Ready
+        pending:
+        - name: etcd-member-ip-10-0-135-168.us-east-2.compute.internal
+          peerURLs: https://etcd-1.sbatsche.devcluster.openshift.com:2380
+          status: Remove
+```
+MemberCondition Matrix:
```

Comment:
For completeness, I'd like to see this table fully expanded with every input and bucket combination considered